src/Runner.Listener/Runner.cs

```diff
@@ -449,7 +453,37 @@ private async Task<int> RunAsync(RunnerSettings settings, bool runOnce = false)
                                 {
                                     Trace.Info($"Received job message of length {message.Body.Length} from service, with hash '{IOUtil.GetSha256Hash(message.Body)}'");
                                     var jobMessage = StringUtil.ConvertFromJson<Pipelines.AgentJobRequestMessage>(message.Body);
-                                    jobDispatcher.Run(jobMessage, runOnce);
+                                    jobDispatcher.Run(Guid.Empty, jobMessage, runOnce);
+                                    if (runOnce)
+                                    {
+                                        Trace.Info("One time used runner received job message.");
+                                        runOnceJobReceived = true;
+                                    }
+                                }
+                            }
+                            // Broker flow
+                            else if (string.Equals(message.MessageType, JobRequestMessageTypes.RunnerJobRequest, StringComparison.OrdinalIgnoreCase))
```

Comment:
@pje fyi this is latest code-path now, tons of other code in this PR needs to be reverted